ui/pages/confirmations/components/confirm/info/shared/gas-fees-section/useFeeCalculations.ts

```diff
@@ -0,0 +1,149 @@
+import { GasFeeEstimates } from '@metamask/gas-fee-controller';
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { Hex } from '@metamask/utils';
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { EtherDenomination } from '../../../../../../../../shared/constants/common';
+import {
+  addHexes,
+  getEthConversionFromWeiHex,
+  getValueFromWeiHex,
+  multiplyHexes,
+} from '../../../../../../../../shared/modules/conversion.utils';
+import { getConversionRate } from '../../../../../../../ducks/metamask/metamask';
+import { useFiatFormatter } from '../../../../../../../hooks/useFiatFormatter';
+import { useGasFeeEstimates } from '../../../../../../../hooks/useGasFeeEstimates';
+import { getCurrentCurrency } from '../../../../../../../selectors';
+import { getGasFeeEstimate } from '../../../../../hooks/utils';
+import { useEIP1559TxFees } from '../../hooks/useEIP1559TxFees';
+import { useSupportsEIP1559 } from '../../hooks/useSupportsEIP1559';
+import { HEX_ZERO } from '../constants';
+
+export function useFeeCalculations(transactionMeta: TransactionMeta) {
+  const currentCurrency = useSelector(getCurrentCurrency);
+  const conversionRate = useSelector(getConversionRate);
+  const fiatFormatter = useFiatFormatter();
+
+  const getFeesFromHex = (hexFee: string) => {
+    const nativeCurrencyFee =
+      getEthConversionFromWeiHex({
+        value: hexFee,
+        fromCurrency: EtherDenomination.GWEI,
+        numberOfDecimals: 4,
+      }) || `0 ${EtherDenomination.ETH}`;
+
+    const currentCurrencyFee = fiatFormatter(
+      Number(
+        getValueFromWeiHex({
+          value: hexFee,
+          conversionRate,
+          fromCurrency: EtherDenomination.GWEI,
+          toCurrency: currentCurrency,
+          numberOfDecimals: 2,
+        }),
+      ),
+    );
+
+    return { currentCurrencyFee, nativeCurrencyFee };
+  };
```

Comment:
It will be more useful to define function above outside hook or use `useCallback`.